apps/champions/lib/champions/units.ex

```diff
@@ -207,6 +217,160 @@ defmodule Champions.Units do
       %CurrencyCost{currency_id: Currencies.get_currency_by_name!("Gems").id, amount: 50}
     ]
 
+  ##########
+  # Fusion #
+  ##########
+
+  @doc """
+  Consume a list of units that meet specific rank and character requirements based on the target
+  unit's rank in order to increase it.
+
+  Returns `{:ok, unit}` or `{:error, reason}`.
+  """
+  def fuse(user_id, unit_id, consumed_units_ids) do
+    with {:unit, {:ok, unit}} <- {:unit, Units.get_unit(unit_id)},
+         {:unit_owned, true} <- {:unit_owned, unit.user_id == user_id},
+         {:can_rank_up, true} <- {:can_rank_up, can_rank_up(unit)},
+         {:unit_not_in_consumed_units, true} <-
+           {:unit_not_in_consumed_units, unit_id not in consumed_units_ids},
+         consumed_units <- Units.get_units_by_ids(consumed_units_ids),
+         {:consumed_units_owned, true} <-
+           {:consumed_units_owned, Enum.all?(consumed_units, &(&1.user_id == user_id))},
+         {:consumed_units_count, true} <-
+           {:consumed_units_count, Enum.count(consumed_units) == Enum.count(consumed_units_ids)},
```

Comment:
Yeah but that's not the only thing it does (in fact, it's not what I put it in for, I'm just realizing it does that), it also handles the case where we receive an id that doesn't exist, since the query we are doing is just "give me all of the units that have one of these ids". If we receive an ID that doesn't exist, we notify that through `{:error, :consumed_units_not_found}`